presto-common/src/main/java/com/facebook/presto/common/type/VarcharType.java

```diff
@@ -23,12 +23,12 @@
 
 import static java.util.Collections.singletonList;
 
-public final class VarcharType
+public class VarcharType
```

Comment:
You probably want to refactor `VarcharType` into `AbstractVarcharType` and `VarcharType` (similar to `AbstractLongType` and `BigintType`) and keep this as final. That way you also avoid the weird need to provide a type signature for the constructor.